homeassistant/components/loqed/config_flow.py

```diff
@@ -2,36 +2,90 @@
 from __future__ import annotations
 
 import logging
-import re
+from typing import Any
 
+import aiohttp
 from loqedAPI import loqed
+import voluptuous as vol
 
+from homeassistant import config_entries
 from homeassistant.components import webhook
 from homeassistant.components.zeroconf import ZeroconfServiceInfo
-from homeassistant.const import CONF_HOST, CONF_WEBHOOK_ID
+from homeassistant.const import CONF_API_TOKEN, CONF_HOST, CONF_WEBHOOK_ID
+from homeassistant.core import HomeAssistant
 from homeassistant.data_entry_flow import FlowResult
-from homeassistant.helpers import config_entry_oauth2_flow
+from homeassistant.exceptions import HomeAssistantError
 from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
 from .const import DOMAIN
 
+_LOGGER = logging.getLogger(__name__)
 
-class ConfigFlow(config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN):
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
     """Handle a config flow for Loqed."""
 
     VERSION = 1
     DOMAIN = DOMAIN
+    _host: str | None = None
 
     @property
     def logger(self) -> logging.Logger:
         """Return logger."""
         return logging.getLogger(__name__)
 
+    async def validate_input(
+        self, hass: HomeAssistant, data: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Validate the user input allows us to connect."""
+
+        newdata = data
+
+        # 1. Checking loqed-connection
+        try:
+            session = async_get_clientsession(hass)
+            res = await session.request(
+                "GET",
+                "https://integrations.production.loqed.com/api/locks/",
```

Comment:
Isn't there a util function for this somewhere ? In other words: Do you need to replicate this code here ?
Normally we leave the API and/or device interactions to an intermediate library.

Reply:
I'll add it to the companion library.

Reply:
Added it to the companion library